src/workflow/orchestrator.py

```
@@ -0,0 +1,298 @@
+        try:
+            if batch_size == 0:
+                # Special case for sync - process all
+                result = worker.process_batch(limit=0)
+            else:
+                result = worker.process_batch(limit=batch_size)
+
+            worker.log_result(result)
+            return result
+
+        except Exception as e:
+            logger.exception(f"Stage {stage} failed with error: {e}")
```

Ruff (0.14.7)

Redundant exception object included in `logging.exception` call

(TRY401)